### DISCUSSION TITLE
Responses to pool

### DISCUSSION THREAD

**Discussion by drksbr** [aldinokemal/go-whatsapp-web-multidevice#216] [General]
I'm facing a problem that is the following: How can I capture the responses for a specific pool?

Ex.: I send a pool within the private message with 4 alternatives, how do I know which alternative the user clicked?

**Comment by aldinokemal** [comment]
Short answer: capturing poll votes is not implemented. Sending a poll works (`POST /send/poll`, `src/ui/rest/send.go:25`), but nothing on the receiving side decodes who voted for what, and that is still the case on v9.0.0.

What happens today: a vote arrives as a regular `events.Message` carrying a `PollUpdateMessage`. It does reach your webhook, but the payload builder has no branch for it (`buildOtherMessageTypes` in `src/infrastructure/whatsapp/event_message.go:391` only handles contact, list, location, live location and order), so you get a `message` event with `id`, `from`, `chat_id`, `timestamp` and no body. There is also no REST endpoint that returns poll results.

Two things are needed to make it useful:

1. Decrypt the vote. whatsmeow already provides this:
   ```go
   if msg.GetPollUpdateMessage() != nil {
       vote, err := client.DecryptPollVote(ctx, evt)
   }
   ```
2. Map the result back to option text. `PollVoteMessage.SelectedOptions` holds SHA-256 hashes of the option strings, not the strings themselves, so you have to hash the options of the original poll (found via `PollUpdateMessage.PollCreationMessageKey`) and compare hashes.

The natural place for it is a new branch in `buildEventPayload` (`src/infrastructure/whatsapp/event_message.go:91`), emitting something like `event: "message.poll_vote"` with the poll message id and the resolved option strings. PR welcome if you want to take it on.
